server/log_rpc_server.go

```diff
@@ -96,21 +96,28 @@ func (t *TrillianLogRPCServer) QueueLeaf(ctx context.Context, req *trillian.Queu
 		return nil, err
 	}
 
-	queueReq := &trillian.QueueLeavesRequest{
-		LogId:  req.LogId,
-		Leaves: []*trillian.LogLeaf{req.Leaf},
+	tree, hasher, err := t.getTreeAndHasher(ctx, req.LogId, optsLogWrite)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Leaf.MerkleLeafHash = hasher.HashLeaf(req.Leaf.LeafValue)
+	if len(req.Leaf.LeafIdentityHash) == 0 {
+		req.Leaf.LeafIdentityHash = req.Leaf.MerkleLeafHash
 	}
-	queueRsp, err := t.QueueLeaves(ctx, queueReq)
+
+	ret, err := t.registry.LogStorage.QueueLeaves(trees.NewContext(ctx, tree), tree, []*trillian.LogLeaf{req.Leaf}, t.timeSource.Now())
```

Comment:
nit: My slight preference is towards shorter lines where feasible. E.g. here you could set `ctx = trees.NewContext(ctx, tree)`, and pass in `ctx`.